ceph/ceph.go

```diff
@@ -178,123 +262,164 @@ func matchSlice(a, b []string) bool {
 
 	return mts, nil
 }
 
 // GetConfigPolicy returns a ConfigPolicy
-func (ceph *Ceph) GetConfigPolicy() (*cpolicy.ConfigPolicy, error) {
-	c := cpolicy.New()
-	socketPath, _ := cpolicy.NewStringRule("socket_path", false, socketPathDefault)
-	socketPrefix, _ := cpolicy.NewStringRule("socket_prefix", false, socketPrefixDefault)
-	socketExt, _ := cpolicy.NewStringRule("socket_ext", false, socketExtDefault)
-	path, _ := cpolicy.NewStringRule("path", false, cephBinPathDefault)
-	p := cpolicy.NewPolicyNode()
-	p.Add(socketPath)
-	p.Add(socketPrefix)
-	p.Add(socketExt)
-	p.Add(path)
-	c.Add([]string{"intel", "storage", "ceph"}, p)
-	return c, nil
+func (ceph *Ceph) GetConfigPolicy() (plugin.ConfigPolicy, error) {
+	policy := plugin.NewConfigPolicy()
+	policy.AddNewStringRule([]string{"intel", "storage", "ceph"}, "socket_path", false, plugin.SetDefaultString(socketPathDefault))
+	policy.AddNewStringRule([]string{"intel", "storage", "ceph"}, "socket_prefix", false, plugin.SetDefaultString(socketPrefixDefault))
+	policy.AddNewStringRule([]string{"intel", "storage", "ceph"}, "socket_ext", false, plugin.SetDefaultString(socketExtDefault))
+	policy.AddNewStringRule([]string{"intel", "storage", "ceph"}, "path", false, plugin.SetDefaultString(cephBinPathDefault))
+	return *policy, nil
 }
 
 // getCephBinaryPath returns path to ceph executable
-func getCephBinaryPath(config map[string]ctypes.ConfigValue) string {
+func getCephBinaryPath(config plugin.Config) (string, error) {
 	// check global config as a first
-	if path, ok := config["path"]; ok {
-		return path.(ctypes.ConfigValueStr).Value
+	path, err := config.GetString("path")
+	if err != nil {
+		return "", err
 	}
 
-	return ""
+	return path, nil
 }
 
 // getCephSocketConf returns path to folder contains daemon-sockets, prefix and extension of socket's name
-func getCephSocketConf(config map[string]ctypes.ConfigValue) Socket {
+func getCephSocketConf(config plugin.Config) (Socket, error) {
 	s := Socket{}
 
 	// set path to socket, defaults to "/var/run/ceph"
-	if path, ok := config["socket_path"]; ok {
-		s.path = path.(ctypes.ConfigValueStr).Value
+	if path, err := config.GetString("socket_path"); err != nil {
+		return Socket{}, err
+	} else {
+		s.path = path
 	}
 
 	// set socket prefix, defaults to "ceph-"
-	if prefix, ok := config["socket_prefix"]; ok {
-		s.prefix = prefix.(ctypes.ConfigValueStr).Value
+	if prefix, err := config.GetString("socket_prefix"); err != nil {
+		return Socket{}, err
+	} else {
+		s.prefix = prefix
 		// if equals "none", set empty socket prefix
 		if strings.ToLower(s.prefix) == "none" {
 			s.prefix = ""
 		}
 	}
 
 	// set socket extension, defaults to "asok"
-	if ext, ok := config["socket_ext"]; ok {
-		s.ext = ext.(ctypes.ConfigValueStr).Value
+	if ext, err := config.GetString("socket_ext"); err != nil {
+		return Socket{}, err
+	} else {
+		s.ext = ext
 	}
 
-	return s
+	return s, nil
+}
+
+// Check is string contains only number
+func isNumber(data string) bool {
```

Comment:
`isNumber` is used in `createNamespace`for which there is no usage. Could you check if those functions are needed?